pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -265,51 +282,26 @@ def _pin_nvrs(self, nvrs: List[str], releases_config) -> Dict:
-        # Make changes
-        releases_yaml_path = build_data_path / "releases.yml"
-        releases_yaml = yaml.load(releases_yaml_path)
-        self._pin_nvrs(nvrs, releases_yaml)
-        yaml.dump(releases_yaml, releases_yaml_path)
-        # Create a PR
         title = f"Pin microshift build for {self.group} {self.assembly}"
         body = f"Created by job run {self.runtime.get_job_run_url()}"
-        match = re.search(r"github\.com[:/](.+)/(.+)(?:.git)?", ocp_build_data_repo_push_url)
-        if not match:
-            raise ValueError(f"Couldn't create a pull request: {ocp_build_data_repo_push_url} is not a valid github repo")
-        head = f"{match[1]}:{branch}"
-        base = self.group
-        if self.runtime.dry_run:
-            self._logger.warning("[DRY RUN] Would have created pull-request with head '%s', base '%s' title '%s', body '%s'", head, base, title, body)
-            d = {"html_url": "https://github.example.com/foo/bar/pull/1234", "number": 1234}
-            result = namedtuple('pull_request', d.keys())(*d.values())
-            return result
-        pushed = await build_data.commit_push(f"{title}\n{body}")
-        result = None
-        if pushed:
-            github_token = os.environ.get('GITHUB_TOKEN')
-            if not github_token:
-                raise ValueError("GITHUB_TOKEN environment variable is required to create a pull request")
-            owner = "openshift"
-            repo = "ocp-build-data"
-            api = GhApi(owner=owner, repo=repo, token=github_token)
-            existing_prs = api.pulls.list(state="open", base=base, head=head)
-            if not existing_prs.items:
-                result = api.pulls.create(head=head, base=base, title=title, body=body, maintainer_can_modify=True)
-            else:
-                pull_number = existing_prs.items[0].number
-                result = api.pulls.update(pull_number=pull_number, title=title, body=body)
-        else:
-            self._logger.warning("PR is not created: Nothing to commit.")
-        return result
+        repo=self.runtime.new_github_client().get_repo("openshift/ocp-build-data")
+        microref=repo.create_git_ref(f"refs/heads/{branch}}",repo.get_git_ref(f"heads/{self.group}").object.sha) # create a microshift branch
+        releasefile=repo.get_contents("releases.yml", ref=branch)
+        yaml=ruamel.yaml.YAML()
+        yaml.preserve_quotes = True
+        releases_yaml = yaml.load(releasefile.decoded_content)
+        self._pin_nvrs(nvrs, releases_yaml) # insert nvrs into release.yaml
+        f=io.StringIO()
+        yaml.dump(releases_yaml,f)
+        f.seek(0)
+        commit=repo.update_file(releasefile.path, "update microshift", f.read(), releasefile.sha, branch=branch) # update releases.yaml in microshift branch
+        pr=repo.create_pull(title=title,body=body,head=f"openshift:{branch}", base={self.group}) # create pr from microshift to {group} branch
```

Comment:
What's the rationale to create pr and then merge immediately? Might as well do a direct commit like we do when creating advisories?

Reply:
I don't think we want `build-microshift` to be the one merging/committing, since we
could be running it manually - it would be ideal for promote job to do that 
- merge PR
- attach build
- attach bugs
- move advisory to qe

Reply:
AFAIK direct commit is not permitted in github repo's setting ? what we do in gen-assembly is also through pr like https://github.com/openshift/ocp-build-data/pull/2613

Reply:
Good point. Merging PRs or attaching builds doesn't seem to be the responsibility of build-microshift job.
Merging PRs could be done by promote-assembly job.
Also Justin proposed another option to include a Brew event in the microshift NVR and adjust Doozer to implicitly include that build without a pin but it is up to you to think about how to implement.

Reply:
if we don't need to pin and just for the release process we can get the build nvr which this job triggered and directly attach to the advisory
or I saw the latest [microshift-4.12.6-202303012057.p0.g50997a2.assembly.4.12.6.el8](https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=2398501) have assembly value in version

Reply:
>if we don't need to pin and just for the release process we can get the build nvr which this job triggered and directly attach to the advisory
If we do like this, the builds will not be part of the assembly. That will violate the design. We need to have all builds managed by assemblies so they can be tracked and rebuilt by our toolings.